### PR TITLE
docs: addition of param and result sections & fix lowest requirements

### LINKED ISSUE #100: installation: problem with cryptography minimum version
As you can see here: https://travis-ci.org/inveniosoftware/invenio-oauthclient/builds/158729080
There is a issue with lowest requirements and python 2.7.

```
CDefError: 'point_conversion_form_t' has no values explicitly defined: refusing to guess which integer type it is meant to be (unsigned/signed, int/long)
```

I think I found the problem and I'll make a PR.

### PR DESCRIPTION
**Pull request by hachreak**
* Improves documentation. (addresses #73)

* Fixes an compatibility issue of cryptography old versions with
  cffi new version. (closes #100)

Signed-off-by: Leonardo Rossi <leonardo.r@cern.ch>

### CODE REVIEW

File: invenio_oauthclient/contrib/cern.py [hunk 1]
```diff
@@ -266,5 +270,8 @@ def account_setup(remote, token, resp):
 
 @identity_loaded.connect
 def on_identity_loaded(sender, identity):
-    """Store groups in session whenever identity changes."""
+    """Store groups in session whenever identity changes.
+
+    :param identity: The user identity where the information are stored.
```
Comment (alizeepace) [inline invenio_oauthclient/contrib/cern.py]: information is stored

File: invenio_oauthclient/contrib/github.py [hunk 2]
```diff
@@ -124,7 +125,32 @@ def _extract_email(gh):
 
 
 def account_info(remote, resp):
-    """Retrieve remote account information used to find local user."""
+    """Retrieve remote account information used to find local user.
+
+    It returns a dictionary with the follow structure:
```
Comment (alizeepace) [inline invenio_oauthclient/contrib/github.py]: following structure

File: invenio_oauthclient/handlers.py [hunk 3]
```diff
@@ -110,7 +144,17 @@ def oauth2_token_setter(remote, resp, token_type='', extra_data=None):
...
                  user=None):
-    """Set token for user."""
+    """Set token for user.
+
+    :param remote: The remote application.
+    :param token: The token to set.
+    :param token_type: The token type. (Default: ``''``)
+    :param extra_data: Extra information. (Default: ``None``)
+    :param user: The user owner of the remote token. If it's not defined,
+        is used automatically the current user. (Default: ``None``)
```
Comment (alizeepace) [inline invenio_oauthclient/handlers.py]: the current user is used automatically

**Comment by hachreak** [comment]
ping @jirikuncar

**Comment by jirikuncar** [comment]
@hachreak LGTM. Please rebase. Thanks

**Comment by hachreak** [comment]
ping @jirikuncar

**Pull request merged**